### PR TITLE
added relational entity to comments

### PR DESCRIPTION
**Pull request by quntao-z**
All comments created should be linked to an entity whether it be a post or a comment.

We can also do a quick look up of the entity field to display comments for a post or other comments

### CODE REVIEW

File: src/main/java/com/bliss/blissapp/Service/CommentsService.java [hunk 1]
```diff
@@ -16,21 +18,21 @@ public class CommentsService {
     private final MongoTemplate mongoTemplate;
     private final CommentsRepository commentsRepository;
 
-    public Optional<Comments> getCommentById(Long id) {
+    public Optional<Comments> getCommentById(UUID id) {
         return commentsRepository.findById(id);
     }
```
Comment (qihongw08) [inline src/main/java/com/bliss/blissapp/Service/CommentsService.java:23]: Should not return type `Optional<Comments>` instead return type `Comments`, unwrap Optional using `.isPresent()` and throw an exception for the controller to handle if comment is not found

File: src/main/java/com/bliss/blissapp/Controller/CommentsController.java [hunk 2]
```diff
@@ -1,44 +1,42 @@
...
         comment.setId(UUID.randomUUID());
+        comment.setEntityId(entityId);
+        comment.setDate(Instant.now());
         commentsService.createComment(comment);
     }
 
     @GetMapping("/{id}")
-    public Optional<Comments> getCommentById(@PathVariable Long id){
+    public Optional<Comments> getCommentById(@PathVariable UUID id){
         return commentsService.getCommentById(id);
```
Comment (qihongw08) [inline src/main/java/com/bliss/blissapp/Controller/CommentsController.java:30]: Method should not return type `Optional<Comments>`, see more details in `CommentsService`

**Review by qihongw08 — CHANGES_REQUESTED** [review]
Resolve these minor problems, everything else looks good!

File: src/main/java/com/bliss/blissapp/Service/CommentsService.java (hunk 1 above)
Comment (quntao-z) [reply src/main/java/com/bliss/blissapp/Service/CommentsService.java:23]: We will use .orElseThrow(()

File: src/main/java/com/bliss/blissapp/Controller/CommentsController.java (hunk 2 above)
Comment (quntao-z) [reply src/main/java/com/bliss/blissapp/Controller/CommentsController.java:30]: It will return comment as we fixed it downstream in the commentService

**Review by qihongw08 — APPROVED** [review]
LGTM!

**Pull request merged**